src/System.Management.Automation/engine/MshMemberInfo.cs

```diff
@@ -3105,6 +3105,26 @@ public PSMemberSet(string name, IEnumerable<PSMemberInfo> members)
             _methods = new PSMemberInfoIntegratingCollection<PSMethodInfo>(this, s_emptyMethodCollection);
         }
 
+        /// <summary>
+        /// Initializes a new instance of PSMemberSet with all the initial members in <paramref name="members"/>.
+        /// This constructor is supposed to be used in TypeTable to reuse the passed-in member collection.
+        /// Null-argument check is skipped here, so callers need to check arguments before passing in.
+        /// </summary>
+        /// <param name="name">Name for the member set.</param>
+        /// <param name="members">Members in the member set.</param>
+        internal PSMemberSet(string name, PSMemberInfoInternalCollection<PSMemberInfo> members)
```

Comment:
It looks to me `_members`, which is exposed by the public property `PSMemberSet.Members` is only populated with data in remoting scenarios (serialization/deserailization).
This change is purely an optimization without changing the behavior of the constructor.

Reply:
It looks to me `_members`, which is exposed by the public property `PSMemberSet.Members` is only populated with data in remoting scenarios (serialization/deserailization).
This change is purely an optimization without changing the behavior of the constructor.

Reply:
I meant that is more more accurately:
```suggestion
        internal PSMemberSet(string name, PSMemberInfoInternalCollection<PSMemberInfo> internalMembers)
```

Reply:
We cannot change the parameter name on the public `PSMemberSet` constructor, since it would be a breaking change -- think about the use of `new PSMemberSet(name: xxx, members: yyy)`.
Given that, I don't want to change the parameter name of the internal constructor too to keep consistent.